extensions-contrib/opentelemetry-extensions/src/test/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReaderTest.java

```diff
@@ -0,0 +1,276 @@
+                            .build());
+
+    metricsDataBuilder.getResourceMetricsBuilder(1)
+            .getInstrumentationLibraryMetricsBuilder(0)
+            .getInstrumentationLibraryBuilder()
+            .setName(INSTRUMENTATION_LIBRARY_NAME)
+            .setVersion(INSTRUMENTATION_LIBRARY_VERSION);
+
+    gaugeMetricBuilder.setName("example_gauge")
+            .getGaugeBuilder()
+            .addDataPointsBuilder()
+            .setAsInt(6)
+            .setTimeUnixNano(TIMESTAMP)
+            .addAttributesBuilder() // test sum with attributes
+            .setKey(METRIC_ATTRIBUTE_FOO_KEY)
+            .setValue(AnyValue.newBuilder().setStringValue(METRIC_ATTRIBUTE_FOO_VAL).build());
+
+    MetricsData metricsData = metricsDataBuilder.build();
+
+    CloseableIterator<InputRow> rows = new OpenTelemetryMetricsProtobufReader(
+            dimensionsSpec,
+            new ByteEntity(metricsData.toByteArray()),
+            "metric.name",
+            null,
+            "descriptor.",
+            "custom."
+    ).read();
+
+    Assert.assertTrue(rows.hasNext());
+    InputRow row = rows.next();
+
+    Assert.assertEquals(4, row.getDimensions().size());
+    assertDimensionEquals(row, "metric.name", "example_sum");
+    assertDimensionEquals(row, "custom.country", "usa");
+    assertDimensionEquals(row, "descriptor.color", "red");
+
+    Assert.assertTrue(rows.hasNext());
+    row = rows.next();
+    Assert.assertEquals(4, row.getDimensions().size());
+    assertDimensionEquals(row, "metric.name", "example_gauge");
+    assertDimensionEquals(row, "custom.env", "devel");
+    assertDimensionEquals(row, "descriptor.foo_key", "foo_value");
+
+  }
+
+  @Test
+  public void testDimensionSpecExclusions()
+  {
+    metricBuilder.setName("example_gauge")
+            .getGaugeBuilder()
+            .addDataPointsBuilder()
+            .setAsInt(6)
+            .setTimeUnixNano(TIMESTAMP)
+            .addAttributesBuilder() // test sum with attributes
+            .setKey(METRIC_ATTRIBUTE_COLOR)
+            .setValue(AnyValue.newBuilder().setStringValue(METRIC_ATTRIBUTE_VALUE_RED).build());
+
+    MetricsData metricsData = metricsDataBuilder.build();
+
+    DimensionsSpec dimensionsSpecWithExclusions = new DimensionsSpec(null,
+            ImmutableList.of(
+                    "descriptor." + METRIC_ATTRIBUTE_COLOR,
+                    "custom." + RESOURCE_ATTRIBUTE_COUNTRY
+            ), null);
+
+    CloseableIterator<InputRow> rows = new OpenTelemetryMetricsProtobufReader(
+            dimensionsSpecWithExclusions,
+            new ByteEntity(metricsData.toByteArray()),
+            "metric.name",
+            null,
+            "descriptor.",
+            "custom."
+    ).read();
+
+    Assert.assertTrue(rows.hasNext());
+    InputRow row = rows.next();
+
+    Assert.assertEquals(2, row.getDimensions().size());
+    assertDimensionEquals(row, "metric.name", "example_gauge");
```

Comment:
again, let's test for all the dimensions that are left to make sure we leaving resources and attribute labels that aren't i the list